### ISSUE TITLE
Access graph canvas context

### ISSUE THREAD

**Issue by nitzanorchestra** [vasturiano/3d-force-graph#384]
**Is your feature request related to a problem? Please describe.**
Is there an option to access graph canvas context, because this gives me null:
const controls = this.graphRef.current.controls();
const canvas = controls.domElement;
const ctx = canvas.getContext('2d');

ctx is null.

thanks.

**Comment by vasturiano** [comment]
@nitzanorchestra this component uses WebGL via ThreeJs for rendering, not a traditional canvas with 2d context.
Can you provide more details on what you are trying to achieve?

**Comment by nitzanorchestra** [comment]
Yes, I want to override the library zoom functionality (scroll wheel) because: 
1. It doesn't provide zoom leveling output. 
2. It doesn't let me control the zoom level with buttons. 

If I use camera position to achieve zoom I have [problem](https://github.com/vasturiano/3d-force-graph/issues/366) when rotating the graph. 

So, I thought to myself, why not work with the canvas itself with "scale" function like [this](http://phrogz.net/tmp/canvas_zoom_to_cursor.html), 
const ctx = canvas.getContext('webgl'); also gives me null.

**Comment by RaulPROP** [comment]
You can use `canvas.getContext("webgl2")` [as shown here](https://codepen.io/raulprop/pen/poyKweW?editors=1111).

**Comment by nitzanorchestra** [comment]
Thanks, it gives me the context now. I see I can't use it as 2d canvas (e.i.: use translate or scale functions there). Do you know a way to make a 2d zoom there, like it was a static image? Or a better way to harness your library zoom?

thanks.

**Comment by RaulPROP** [comment]
I'm not the library owner, but I have no idea.

**Comment by vasturiano** [comment]
@nitzanorchestra you should use the camera [trackball controls](https://threejs.org/docs/#examples/en/controls/TrackballControls), accessible via `myGraph.controls()`.

**Comment by nitzanorchestra** [comment]
Thanks, will try and update. 
Will methods there provide me those, or I should still try to implement webgl2 zooming:
1. zoom level output?
2. control the zoom level with buttons? (plus, minus by some percent).
3. zoom according to mouse pointer?

**Comment by nitzanorchestra** [comment]
@vasturiano that what I did in the first message 'this.graphRef.current.controls();' no?

**Comment by vasturiano** [comment]
@nitzanorchestra not sure precisely what you mean, but another thing you could consider is using the lower level [three-forcegraph] component, and building the ThreeJS scene scaffolding custom to your application.
That will give you complete freedom on how you manipulate the camera position, pointer interaction, etc.
